File: tests/fixtures/python/deep_nesting.py
"""Function with deeply nested control structures (depth > 4)."""


def deeply_nested(data):
    for group in data:
        if group.active:
            for item in group.items:
                if item.valid:
                    for sub in item.children:
                        if sub.ready:
                            process(sub)
    return True


def moderately_nested(data):
    for item in data:
        if item.active:
            for sub in item.children:
                process(sub)
    return True


def process(x):
    pass

File: tests/fixtures/go/deep_nesting.go
package main

func DeeplyNested(data [][][]int) bool {
	for _, group := range data {
		if len(group) > 0 {
			for _, items := range group {
				if len(items) > 0 {
					for _, val := range items {
						if val > 0 {
							process(val)
						}
					}
				}
			}
		}
	}
	return true
}

func ModeratelyNested(data []int) bool {
	for _, item := range data {
		if item > 0 {
			process(item)
		}
	}
	return true
}

func process(x int) {}
